src/elements/Pagination/index.jsx

```diff
@@ -0,0 +1,205 @@
+      }
+    } else {
+      steps = this.calculatePreviousSteps(leftLength, currentPage)
+        .concat(currentPage, this.calculateFollowingSteps(rightLength, currentPage, totalPages))
+    }
+    return steps
+  }
+
+  calculatePreviousSteps (leftLength, currentPage) {
+    const res = []
+
+    if ((currentPage - 1) > leftLength) {
+      for (let i = 0; i < leftLength; i += 1) {
+        if (i === leftLength - 1) {
+          res.unshift(1)
+        } else if (i === leftLength - 2) {
+          res.unshift('ellipsis')
+        } else {
+          res.unshift(currentPage - i - 1)
+        }
+      }
+    } else {
+      for (let i = 0; i < leftLength; i += 1) {
+        res.push(i + 1)
+      }
+    }
+    return res;
+  }
+
+  calculateFollowingSteps (rightLength, currentPage, totalPages) {
+    const res = []
+
+    if ((totalPages - currentPage) > rightLength) {
+      for (let i = 0; i < rightLength; i += 1) {
+        if (i === rightLength - 1) {
+          res.push(totalPages)
+        } else if (i === rightLength - 2) {
+          res.push('ellipsis')
+        } else {
+          res.push(currentPage + i + 1)
+        }
+      }
+    } else {
+      for (let i = 0; i < rightLength; i += 1) {
+        res.push(currentPage + i + 1)
+      }
+    }
+    return res;
+  }
+
+  renderPagination () {
+    const { currentPage, totalPages } = this.state
+    if (currentPage < 1) {
+      this.changePage(1)
+    } else if (currentPage > totalPages) {
+      this.changePage(totalPages)
+    }
+
+    const paginations = []
+    const steps = this.calculateSteps();
+
+    for (let i = 0; i < steps.length; i += 1) {
+      const label = steps[i];
+      let isClickable = true;
+
+      if (currentPage === steps[i]) {
+        isClickable = false;
+      }
+      if (label === 'ellipsis') {
+        paginations.push(<PaginationLink isPageBreak={true} key={i} />)
+      } else {
+        paginations.push(
+          <PaginationLink
+            label={label}
+            isClickable={isClickable}
+            onClick={(page) => { this.onClick(page) }}
+            key={i} />
+        )
+      }
+    }
+
+    return paginations;
+  }
+
+  renderLeftCaret () {
+    return (
+      <li className="Pagination__caret">
+        { this.state.currentPage <= 1
+          ? <span>{'<'}</span>
+          : <a onClick={() => {
+            this.onClick(this.state.currentPage - 1)
+          }}>
+            {'<'}
+          </a> }
+      </li>
+    )
+  }
+
+  renderRightCaret () {
+    return (
+      <li className="Pagination__caret">
+        { this.state.currentPage >= this.state.totalPages
+          ? <span>{'>'}</span>
+          : <a onClick={() => {
+            this.onClick(this.state.currentPage + 1)
+          }}>
+            {'>'}
+          </a>
+        }
+      </li>
+    )
+  }
+
+  render () {
+    const leftCaret = this.renderLeftCaret();
+    const rightCaret = this.renderRightCaret();
+    const pagination = this.renderPagination();
```

Comment:
do we really need to assign this instances to variable?

Reply:
You are right, we can directly use these functions in JSX tag.